src/views/Recorder/RecentURLs.tsx

```diff
@@ -0,0 +1,112 @@
+import { css } from '@emotion/react'
+import { Badge, Reset, Tooltip } from '@radix-ui/themes'
+import { XIcon } from 'lucide-react'
+
+interface RecentURLsProps {
+  urls: string[]
+  disabled: boolean
+  onSelectURL: (url: string) => void
+  onRemoveURL: (url: string) => void
+}
+
+export function RecentURLs({
+  urls,
+  disabled,
+  onSelectURL,
+  onRemoveURL,
+}: RecentURLsProps) {
+  if (urls.length === 0) {
+    return null
+  }
+
+  return (
+    <ul
+      aria-label="Recent URLs"
+      css={css`
+        list-style: none;
+        padding: 0;
+        margin: 0 0 var(--space-3);
+        display: flex;
+        flex-wrap: wrap;
+        gap: var(--space-1) var(--space-3);
+      `}
+    >
+      {urls.map((url) => (
+        <li
+          key={url}
+          css={css`
+            max-width: 100%;
+          `}
+        >
+          <Badge
+            size="1"
+            variant="soft"
+            color="gray"
+            highContrast
+            css={css`
+              position: relative;
+              max-width: 100%;
+
+              &:hover {
+                background-color: var(--gray-4);
+              }
+
+              &:focus-within > button:last-of-type,
+              &:hover > button:last-of-type {
+                opacity: 1;
+              }
+            `}
+          >
+            <Reset>
+              <button
+                type="button"
+                aria-label="Select URL"
+                disabled={disabled}
+                css={css`
+                  flex: 1;
+                  overflow: hidden;
+                  text-overflow: ellipsis;
+                  white-space: nowrap;
```

Comment:
Takes care of really long URLs even when the window size is small:
<img width="433" alt="image" src="https://github.com/user-attachments/assets/e11eec21-f462-40fc-b16d-f5702fe21b32" />